pages/docs/deployment/self-host-v3.mdx

```diff
@@ -0,0 +1,350 @@
+---
+description: Step-by-step guide to run Langfuse on your local machine using docker compose.
+---
+
+import { Callout } from "nextra/components";
+
+# Self-hosted deployment - V3 Preview - Open Source LLM Observability
+
+<Callout type="warning">
+  This guide covers a developer preview which is **not suitable for production use**.
+  V3 is under active development and we plan to ship a production-ready version by the end of November 2024.
+  We share this information to gather feedback from our awesome developer community.
+
+  For a production-ready setup, follow the [self-hosting guide](/docs/deployment/self-host)
+  or consider using [Langfuse Cloud](https://cloud.langfuse.com) maintained by the Langfuse team.
+
+  If you are on a V2 setup and want to update your stack, take a look at our [migration guide](/docs/deployment/migrate-v2-to-v3).
+</Callout>
+
+## Components
+
+Langfuse consists of multiple storage components and two Docker containers:
+
+- **Langfuse Web**: The main web application serving the Langfuse UI and APIs.
+- **Langfuse Worker**: A worker that asynchronously processes events.
+- **Postgres**: The main database for transactional workloads.
+- **Redis**: A fast in-memory data structure store. Used for queue and cache operations.
+- **S3/Blob Store**: Object storage to persist all incoming events, multi-modal inputs, and large exports.
+- **Clickhouse**: High-performance OLAP database which stores traces, observations, and scores.
+
+[//]: # (TODO: Update architecture diagram to include all components above)
+For a detailed component and architecture diagram, refer to [CONTRIBUTING.md](https://github.com/langfuse/langfuse/blob/main/CONTRIBUTING.md#architecture-overview).
+
+### Postgres Database
+
+Langfuse requires a persistent Postgres database to store its state.
+You can use a managed service on AWS, Azure, or GCP, or host it yourself.
+At least version 12 is required.
+
+### Redis
+
+Langfuse uses Redis for caching and queue operations.
+You can use a managed service on AWS, Azure, or GCP, or host it yourself.
+At least version 7 is required and the instance must have `maxmemory-policy=noeviction` configured.
+You may use Valkey instead of Redis.
```

Comment:
I did a small smoke test when experimenting on the helm chart. Valkey 8 worked just fine and is also API compatible with standard redis according to the maintainers. From my side the confidence would be >90% that it can be an easy drop-in replacement for new stacks.

Reply:
Sounds good, does it make sense to include it in the helm chart and e2e tests? I'd imagine that this is a common concern otherwise and we might run into issues where we do not test against it while many people might use it

Reply:
I'll track this in https://linear.app/langfuse/issue/LFE-2835/add-valkey-deployment-option-to-e2e-tests-in-langfuse-repo to add valkey tests around GA.